lib/spack/spack/build_systems/python.py

```diff
@@ -136,31 +137,50 @@ def view_file_conflicts(self, view, merge_map):
         return conflicts
 
     def add_files_to_view(self, view, merge_map, skip_if_exists=True):
-        if not self.extendee_spec:
+        # Patch up shebangs to the python linked in the view only if python is built by Spack.
+        if not self.extendee_spec or self.extendee_spec.external:
             return super().add_files_to_view(view, merge_map, skip_if_exists)
 
+        # We only patch shebangs in the bin directory.
+        copied_files: Dict[Tuple[int, int], str] = {}  # File identifier -> source
+        delayed_links: List[Tuple[str, str]] = []  # List of symlinks from merge map
+
         bin_dir = self.spec.prefix.bin
         python_prefix = self.extendee_spec.prefix
-        python_is_external = self.extendee_spec.external
-        global_view = fs.same_path(python_prefix, view.get_projection_for_spec(self.spec))
         for src, dst in merge_map.items():
-            if os.path.exists(dst):
+            if skip_if_exists and os.path.lexists(dst):
                 continue
-            elif global_view or not fs.path_contains_subdirectory(src, bin_dir):
+
+            if not fs.path_contains_subdirectory(src, bin_dir):
```

Comment:
Do we have weird cases where a script with a shebang is in another place than `spec.prefix.bin`?

Reply:
I'm not sure... I believe the original implementation was to only look in `bin/*` expecting scripts there. Then a second iteration of it was to limit it to executable, regular files that at least have a shebang...

I guess the latter could be the _only_ criterion, given that on Windows `bin_dir == prefix`.

But that is also costly again especially for `python` itself, given that it has 8000+ files...

Reply:
Ok, let's get this in as is, and think how to deal with edge cases when they happen.